datafusion/sql/src/unparser/expr.rs

```diff
@@ -958,8 +959,16 @@ impl Unparser<'_> {
             }
             DataType::Float32 => Ok(ast::DataType::Float(None)),
             DataType::Float64 => Ok(ast::DataType::Double),
-            DataType::Timestamp(_, _) => {
-                not_impl_err!("Unsupported DataType: conversion: {data_type:?}")
+            DataType::Timestamp(time_unit, _) => {
+
+                let time_precision = match time_unit {
```

Comment:
Not 100% sure time_precision should be specified. Seems adding precision is the right thing as we have this info.
Not all precision is supported by target engines, for example  despite the following sql works in Postgres `select CAST('2013-08-22 06:24:12-02' AS TIMESTAMP(9))` I see `TIMESTAMP(9) precision reduced to maximum allowed, 6` in Postgres logs. Also tested max precision in Dremio, Snowflake - works fine.